### PR TITLE
[Flutter-Parent][MBL-13495] Course grades list

### PR DESCRIPTION
**Pull request by CalvinKern**

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1013554&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/lib/utils/design/parent_theme.dart [hunk 1]
```diff
@@ -208,22 +208,25 @@ class _ParentThemeState extends State<ParentTheme> {
...
+      accentColorBrightness: isDarkMode ? Brightness.light : Brightness.dark,
+      textTheme: textTheme,
+      primaryTextTheme: isDarkMode ? textTheme : _buildTextTheme(Colors.white, fadeColor: Colors.white70),
+      accentTextTheme: isDarkMode ? textTheme : _buildTextTheme(Colors.white, fadeColor: Colors.white70),
+      iconTheme: IconThemeData(color: onSurfaceColor),
+      primaryIconTheme: IconThemeData(color: isDarkMode ? ParentTheme.tiara : Colors.white),
+      accentIconTheme: IconThemeData(color: isDarkMode ? Colors.black : Colors.white),
+      dividerColor: isHC ? onSurfaceColor : isDarkMode ? nearSurfaceColor : ParentTheme.tiara,
+      buttonTheme: ButtonThemeData(height: 48, minWidth: 120),
+      splashColor: studentColor.withOpacity(.12),
```
Comment (JordanMarshall) [inline apps/flutter_parent/lib/utils/design/parent_theme.dart]: Won't this cause ripple feedback to effectively be invisible on themed elements like FABs and tab bars?

File: apps/flutter_parent/lib/utils/common_widgets/empty_panda_widget.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const EmptyPandaWidget({Key key, this.svgPath, this.title, this.subtitle}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return FullScreenScrollContainer(
+      children: <Widget>[
+        if (svgPath != null) SvgPicture.asset(svgPath, excludeFromSemantics: true),
+        if (svgPath != null && (title != null || subtitle != null)) SizedBox(height: 64),
+        if (title != null)
+          Text(
```
Comment (JordanMarshall) [inline apps/flutter_parent/lib/utils/common_widgets/empty_panda_widget.dart]: Let's get some horizontal padding on these text widgets. I'd say at least 16 - although imo 32 looks better for this type of content - but Dave would have the final say.

File: apps/flutter_parent/lib/screens/courses/details/course_grades_screen.dart [hunk 3]
```diff
@@ -13,20 +13,289 @@
...
+  String _courseGrade(BuildContext context, CourseGrade grade) {
+    final format = NumberFormat();
+    format.maximumFractionDigits = 2;
+
+    if (grade.noCurrentGrade()) {
+      return L10n(context).noGrade;
+    } else {
+      return grade.currentGrade()?.isNotEmpty == true
+          ? grade.currentGrade()
+          : '${format.format(grade.currentScore())}%';
```
[the remark this answers is not in the document]
Comment (CalvinKern) [reply apps/flutter_parent/lib/screens/courses/details/course_grades_screen.dart]: Oh nice to know that the percent one is built in. I'll get that changed.

File: apps/flutter_parent/lib/models/assignment.dart [hunk 4]
```diff
@@ -128,4 +133,67 @@ abstract class Assignment implements Built<Assignment, AssignmentBuilder> {
...
+    } else {
+      return SubmissionStatus.SUBMITTED;
+    }
+  }
+
+  // Returns true if the submission is marked as missing, or if it's pass due and either no submission or 'fake' submission
+  bool _isMissingSubmission() {
+      if (submission?.missing == true) return true;
+
+    final isPassDue = dueAt?.isBefore(DateTime.now()) == true;
```
[the remark this answers is not in the document]
Comment (CalvinKern) [reply apps/flutter_parent/lib/models/assignment.dart]: lol, total brain fart moment. Nice catch.

File: apps/flutter_parent/lib/utils/design/parent_theme.dart (hunk 1 above)
Comment (CalvinKern) [reply apps/flutter_parent/lib/utils/design/parent_theme.dart]: Yuuup, you're totally right. Guess we're just going to have things with gray selectable by default.

File: apps/flutter_parent/lib/screens/courses/details/course_details_model.dart [hunk 5]
```diff
@@ -22,27 +26,68 @@ class CourseDetailsModel extends BaseModel {
...
       // Await the results
       course = await courseFuture;
-      assignmentGroups = await assignmentsFuture;
     });
   }
 
+  // observed_users doesn't work right now on the groups endpoint, so we have to pull submissions manually
+  // TODO: Remove everything except the loadAssignmentGroups call (and removing unpublished) once LA-274 is implemented
+  Future<List<AssignmentGroup>> _loadAssignments({bool forceRefresh = false}) {
+//    if (true) return Future.error("fail plz");
```
[the remark this answers is not in the document]
Comment (CalvinKern) [reply apps/flutter_parent/lib/screens/courses/details/course_details_model.dart]: 💀

File: apps/flutter_parent/lib/utils/common_widgets/empty_panda_widget.dart (hunk 2 above)
Comment (CalvinKern) [reply apps/flutter_parent/lib/utils/common_widgets/empty_panda_widget.dart]: Talked with the Davester.... He said 32 would be good, so I'll push that up!

**Pull request merged**